src/dataStructures/binarySearchTree.js

```diff
@@ -39,6 +26,19 @@ class BinarySearchTree {
     }
   }
 
+  // Private method
+  static addNodeHelper(value, head) {
```

Comment:
If we make this static, then it will be visible to the class API we provide. Better idea to leave this outside so only the class can access this in the current file and it does not get forwarded with the class methods.